### DISCUSSION TITLE
Adding  nodeSelector/affinity to Job Pods in kubernetes mode

### DISCUSSION THREAD

**Discussion by knkarthik** [actions/actions-runner-controller#3007] [Questions] [answered]
Currently, I'm adding `nodeSelector` and `affinity` to `template.spec` in [gha-runner-scale-set](https://github.com/actions/actions-runner-controller/blob/master/charts/gha-runner-scale-set/values.yaml#L109) chart. This works well and the runner pods are going where I want them to go. However, when I use `jobs.container` in my actions I want the `*-workflow` job pods created to colocate with the runner pod that created it. My questions are:

1. Is the `runner` pod and the job pod guaranteed to run on the same node?
2. Does the `nodeSelector` /`affinity` applied to runner also apply to the job pod?
3. If not, how can I make sure that these 2 run on the same node?

I _think_  I've to add `nodeSelector` and `affinity` to the job pod by [leveraging hooks extension]( https://github.com/actions/actions-runner-controller/issues/2992#issuecomment-1764855221) somehow.

**Comment by nikola-jokic** [comment, marked as answer]
Hey @knkarthik,

1. Runner pod and the job pod should land on the same node. It is explicitly set in the hook [here](https://github.com/actions/runner-container-hooks/blob/496287d61d08cb25cbab761aab8671dea244338c/packages/k8s/src/k8s/index.ts#L89)
2. It does not. Job pod does not rely on the runner pod spec.
3. I hope it is answered by the first point :relaxed:

**Reply by knkarthik** [reply]
Sweet!! Thanks for the link.
